homeassistant/components/just_nimbus/sensor.py

```diff
@@ -0,0 +1,180 @@
+        name="Drink Flow",
+        icon="mdi:water-pump",
+        native_unit_of_measurement=VOLUME_FLOW_RATE_LITERS_PER_MINUTE,
+        unique_id="jn:drink_flow",
+    ),
+    JustNimbusEntityDescription(
+        key="pump_pressure",
+        name="Pump Pressure",
+        icon="mdi:arrow-collapse-vertical",
+        native_unit_of_measurement=PRESSURE_BAR,
+        device_class=SensorDeviceClass.PRESSURE,
+        unique_id="jn:pump_pressure",
+    ),
+    JustNimbusEntityDescription(
+        key="pump_starts",
+        name="Pump Starts",
+        icon="mdi:restart",
+        unique_id="jn:pump_starts",
+    ),
+    JustNimbusEntityDescription(
+        key="pump_hours",
+        name="Pump Hours",
+        icon="mdi:clock",
+        device_class=SensorDeviceClass.DURATION,
+        unique_id="jn:pump_hours",
+    ),
+    JustNimbusEntityDescription(
+        key="reservoir_temp",
+        name="Reservoir Temperature",
+        icon="mdi:thermometer-water",
+        native_unit_of_measurement=TEMP_CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        unique_id="jn:reservoir_temp",
+    ),
+    JustNimbusEntityDescription(
+        key="reservoir_content",
+        name="Reservoir Content",
+        icon="mdi:car-coolant-level",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:reservoir_content",
+    ),
+    JustNimbusEntityDescription(
+        key="total_saved",
+        name="Total Saved",
+        icon="mdi:water-opacity",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:total_saved",
+    ),
+    JustNimbusEntityDescription(
+        key="total_replenished",
+        name="Total Replenished",
+        icon="mdi:water",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:total_replenished",
+    ),
+    JustNimbusEntityDescription(
+        key="error_code",
+        name="Error Code",
+        icon="mdi:bug",
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="",
+        unique_id="jn:error_code",
+    ),
+    JustNimbusEntityDescription(
+        key="totver",
+        name="Total Use",
+        icon="mdi:chart-donut",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:totver",
+    ),
+    JustNimbusEntityDescription(
+        key="reservoir_content_max",
+        name="Max Reservoir Content",
+        icon="mdi:waves",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:reservoir_content_max",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the JustNimbus sensor."""
+    coordinator: JustNimbusCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        JustNimbusSensor(
+            client=justnimbus.JustNimbusClient(entry.data[CONF_CLIENT_ID]),
+            entry_id=entry.entry_id,
+            device_id=entry.data[CONF_CLIENT_ID],
```

Comment:
This client ID seems to be part of the device authentication as well? Does it ever change? Isn't this more of a connection identifier and not a device identifier?

Reply:
> This means that the API isn't very well protected, but since it doesn't contain any privileged data

Uh, that would be a CVE if that is indeed unprotected, as behavior can be extracted from this data. For example, one could track if someone is home.

Reply:
Btw, the main concern is, can a user change it? E.g., if you re-register it? or any action at all?

Reply:
> > This means that the API isn't very well protected, but since it doesn't contain any privileged data
> 
> Uh, that would be a CVE if that is indeed unprotected (IMHO), as behavior can be extracted from this data. For example, one could track if someone is home.

I agree with you that this is far from optimal, however since I don't have any relation to JustNimbus I can't do anything about it.  By default, a third party doesn't know which ID belongs to which address.

> Btw, the main concern is, can a user change it? E.g., if you re-register it? or any action at all?

As far as I know, the only one that can change this is the guy running JustNimbus.